DKTDDSwiftUI/Greetings.swift

```diff
@@ -8,84 +8,43 @@
 import Foundation
 
 class Greetings {
-  
-    func handleNamesCountEqualsTwo(_ names: inout [String], _ result: inout String) {
-        if names[0].contains(",") {
-            var newNames = names[0].components(separatedBy: ", ")
-            result += "\(newNames[0]), \(newNames[1]), and \(names[1])"
-        } else if names[1].contains(","){
-            var newNames = names[1].components(separatedBy: ", ")
-            result += "\(names[0]), \(newNames[0]), and \(newNames[1])"
-        } else {
-            result += "\(names[0]), and \(names[1])"
-        }
-    }
-    
-    func handlesArraysOfNames(_ names: [String], _ lastName: String, _ result: inout String, _ nameWithComas: inout [String] ) {
-        var names = nameWithComas.isEmpty ? names : nameWithComas
-        for name in names {
-            checkeNameWithComasOrDoubleQuotes(name, &nameWithComas)
-            if names.count == 1 && !nameWithComas.isEmpty {
-                result += "\(names[0]), and \(names[1])"
-            }
-            
-            if names.count == 2 {
-                handleNamesCountEqualsTwo(&names, &result)
-                break
-            }
-            
-            if lastName == name {
-                result += "and \(name)"
-            } else if name.uppercased() == name {
-                result += "\(name), "
-            } else if lastName.uppercased() == name {
-                result += "and \(name)"
-            } else {
-                result += "\(name), "
-            }
-            
-        }
-    }
+    // MARK: Properties
+    var result = ""
+    var nameWithCommas = [String]()
+    var lastName = ""
+    var namesString = ""
```

Comment:
I think this is a larger issue with creating a class in this instance. When we asked for the differences between a class vs. a struct vs. an enum, we wanted to see how you would create this object if this were a real world application. One of the big advantages of a struct or enum is removing unnecessary state. Objects should have limited control and knowledge of the objects around them. These public properties with mutating state make me nervous. I wouldn't want another developer to accidentally change the state or think the state is fair game to change.

Reply:
I couldn't agree more with you. Structs also have the benefit of mutation in safety and you can trust no other part in the app is changing the data at the same time making it very helpful in multi-threading.
Let me make this object a struct instead. Thanks 🙏